forwarder/forwarder/forwarderobject.py

```diff
@@ -151,10 +166,36 @@ def handle_app_update(self, task_id, future):
             print(f"*** TASK RETURN SUCCEEDED: {task_id} -- {time.time()}***")
             logger.info(f"Task:{task_id} succeeded")
 
-    def heartbeat_fail(self):
+    def _endpoint_heartbeat_fail(self):
         """Return true if too many heartbeats have been missed"""
         return int(time.time() - self.executor.last_response_time) > \
                     (self.max_heartbeats_missed * self.heartbeat_threshold)
+    
+    def _heartbeat_loop(self, kill_event):
+        while not kill_event.is_set():
+            logger.info("Activating executor.send_heartbeat()")
+            self.executor.send_heartbeat()
+            time.sleep(self.heartbeat_threshold)
+
+    def _task_status_loop(self, kill_event):
+        while not kill_event.is_set():
+            logger.debug("Pulling from task_status_q")
+            try:
+                task_status_delta = self.task_status_q.get(timeout=1)
```

Comment:
Maybe set timeout = `heartbeat_period` and when the queue.Empty exception fires continue.

Reply:
sure, cleaner with fewer sleeps